onnxruntime/core/providers/acl/nn/pool.cc

```diff
@@ -37,17 +37,17 @@ ACLNEPool PoolOperation(onnxruntime::OpKernelContext* context,
   const Tensor* X = context->Input<Tensor>(0);
   const TensorShape& x_shape = X->Shape();
 
-  std::vector<int64_t> pads = pool_attrs.pads;
-  std::vector<int64_t> strides = pool_attrs.strides;
-  std::vector<int64_t> kernel_shape = pool_attrs.kernel_shape;
+  auto pads = pool_attrs.pads;
```

Comment:
Do you need to make a copy of the object? If not. it should be auto& or const auto&

Reply:
I don't know if it's needed, though in the original code those were copies so I kept them as they were.

Reply:
You can just make them const auto& and see if it compiles

Reply:
Doesn't compile, but I dont think that would be a good way to check if this is safe to do,as we could end up accidentally modifying the `pads` value of pool_attrs instead of working on a copy of it. As I said, I don't know wnough of how this actually work to know which is the right way, so I'll just stick to what the original semantics were and fix the typing issues